students/johnpharmd/lesson04/trigrams.py

```diff
@@ -28,30 +28,86 @@
 
 # Data
 sample_st = 'I wish I may I wish I might'
+read_data = []
 s1 = ''
 s2 = ''
+st_key = ''
 s3 = ''
-count = 0
-st_set = set()
 st_dict = {}
+count = 0
 pron_set = set(['I', 'you', 'she', 'he', 'it', 'we', 'they'])
-
+n = 0
+new_st_lst= []
+testfile_dict = {1: 'sherlock_small.txt', 2: 'sherlock.txt'}
 
 # Processing
-st_split_lst = sample_st.split(' ')
-print('st_split_lst: ', st_split_lst, '\n')
-for i, word in enumerate(st_split_lst):
-    if i + 2 < len(st_split_lst):
-        s1 = word
-        s2 = st_split_lst[i + 1]
-        s3 = st_split_lst[i + 2]
-        st_set.update([s1, s2])
-        st_dict[count] = {frozenset(st_set): s3}
-        # print('st_set:', st_set)
-        count += 1
-        s1 = ''
-        s2 = ''
-        s3 = ''
-        st_set = set()
-for k, v in st_dict.items():
-    print(k, v, '\n')
+def clean_data(choice=1):
+    # use 2 as param, if want to test with 'sherlock.txt'
+    with open(testfile_dict[choice], 'r') as f:
+    # header_size = 0
+    # with open('sherlock.txt') as f:
+        read_data = f.read()
+        # read_data = f.seek('To Sherlock Holmes she')
+    read_data = ''.join(read_data.split('\r'))
+    read_data = ''.join(read_data.split('"'))
+    read_data = ' '.join(read_data.split('\n'))
+    read_data = ' '.join(read_data.split('--'))
+    read_data = ' '.join(read_data.split('-'))
+    read_data = ' '.join(read_data.split(', '))
+    read_data = ' '.join(read_data.split('. '))
+    read_data = ''.join(read_data.split('('))
+    read_data = ''.join(read_data.split(')'))
+    read_data = read_data.strip('.')
+    read_data = read_data.strip('  ')
+    read_data = read_data.split(' ')
+
+    print('type(read_data):', type(read_data), '\n')
+    # print('read_data:', read_data)
+    print(read_data)
+    return read_data
+
+
+def make_st_dict(lst):
+# lines between this and next comment line can be encapsulated into a
+# separate fxn
+# possible name: def make_st_dict(lst):
+    for i, word in enumerate(read_data):
+        if i + 2 < len(read_data):
+            s1 = word
+            s2 = read_data[i + 1]
+            st_key = (s1, s2)
+            s3 = read_data[i + 2]
+            if st_key not in st_dict:
+                st_dict[st_key] = [s3]
+            else:
+                st_dict[st_key] += [s3]
+        # count += 1
+    print(st_dict)
+# see comment line above
```

Comment:
the indentation is off in this function, also your comment should have one indent from function definition